### PR TITLE
[IMP] account,l10n_de: GoBD improvements

### PR DESCRIPTION
**Pull request by antoine162**
* In standard: enable tracking on partner name
* In l10n_de: enable tracking on SteuerNr.
* Constraint on account : "You can not change the code/name of an account if it contains hashed entries."
* enable audit trail by default when loading a new German CoA

taskid: 3950840

### CODE REVIEW

File: addons/l10n_de/models/chart_template.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ def _get_de_res_company(self):
             self.env.company.id: {
                 'external_report_layout_id': 'l10n_din5008.external_layout_din5008',
                 'paperformat_id': 'l10n_din5008.paperformat_euro_din',
+                'check_account_audit_trail': True,
```
Comment (gawa-odoo) [inline addons/l10n_de/models/chart_template.py:15]: Shouldn't it be in the new module?
If you install the new module, it's not checked. I don't know if it's intentional

Comment (gawa-odoo) [reply addons/l10n_de/models/chart_template.py:15]: So, the logic, is that even if we don't have the new module, but that we have already audit trail installed, we want to activate it for new companies.

**Review by gawa-odoo — COMMENTED** [review]
@robodoo r+

**Comment by gawa-odoo** [comment]
@robodoo rebase-ff

**Comment by antoine162** [comment]
@robodoo retry

File: addons/l10n_de/models/account_account.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    def write(self, vals):
+        if 'DE' in self.company_id.account_fiscal_country_id.mapped('code') and ('code' in vals or 'name' in vals):
+            hashed_aml_domain = [
+                ('account_id', 'in', self.ids),
+                ('move_id.inalterable_hash', '!=', False),
+            ]
+            if self.env['account.move.line'].search_count(hashed_aml_domain, limit=1):
+                raise UserError(_("You can not change the code/name of an account if it contains hashed entries."))
+        super().write()
```
Comment (gawa-odoo) [inline addons/l10n_de/models/account_account.py:16]:
```suggestion
        super().write(vals)
```

Comment (antoine162) [reply addons/l10n_de/models/account_account.py:16]: XD thanks

Comment (gawa-odoo) [reply addons/l10n_de/models/account_account.py]: On l'a loupé tous les 2 :smile:

**Comment by gawa-odoo** [comment]
@robodoo r+

**Comment by FalcoBolger** [comment]
@antoine162 The extension of `write` on `account.account` is missing returning the result of the call to `super`, see [`15b1c05` (#167973)](https://github.com/odoo/odoo/pull/167973/commits/15b1c0586ac4ae88926c3fd50f47754e9c426119#diff-14762683af1054df15909d1eed0bec383eca32ee16898d61350825736759a035R16).
The runbot in OdooSH fails on the test `/base:TestOverrides.test_writes`.

**Comment by antoine162** [comment]
> test_writes



> @antoine162 The extension of `write` on `account.account` is missing returning the result of the call to `super`, see [`15b1c05` (#167973)](https://github.com/odoo/odoo/pull/167973/commits/15b1c0586ac4ae88926c3fd50f47754e9c426119#diff-14762683af1054df15909d1eed0bec383eca32ee16898d61350825736759a035R16). The runbot in OdooSH fails on the test `/base:TestOverrides.test_writes`.

Thanks - I opened https://github.com/odoo/odoo/pull/172524 to fix that.

**Pull request closed without merging**